src/partielspy/plugin_list.py

```diff
@@ -0,0 +1,43 @@
+import json
+
+from .plugin import Plugin
+from .plugin_key import PluginKey
+
+
+class PluginList(list[Plugin]):
+    @classmethod
+    def from_json(cls, data: json):
+        plugin_list = cls()
+        for plugin in data:
+            key = plugin.get("key", "")
+            if not isinstance(key, dict):
+                raise TypeError("The key of the plugin is not a dictionary")
+            plugin_key = PluginKey(
+                identifier=key.get("identifier", ""),
+                feature=key.get("feature", ""),
+            )
+            plugin_list.append(Plugin(plugin_key))
+        return plugin_list
+
+    def to_json(self) -> list[dict]:
+        return json.dumps([plugin._to_json() for plugin in self], indent=4)
+
+    def get(self, identifier: str, feature: str) -> Plugin | None:
+        for plugin in self:
+            key = plugin.key
+            if key.identifier == identifier and key.feature == feature:
+                return plugin
+        return None
+
+    def find(self, needle: str) -> "PluginList":
```

Comment:
The method looks for all the plugins with the identifier or feature that matches the "needle". The term find led me to think that it tries to find one plugin. I would rename the method with a more explicit name, such as "get matching plugins", and the term "needle" with "term to look for".